### PR TITLE
Fix leaks

### LINKED ISSUE #76: Remove Data.tsx file
This file is no longer neeeded because we made use of the following tool:

https://github.com/typicode/json-server#getting-started

make sure to run the following commands

`npm install -g json-server`

and

`json-server --watch db.json`

### LINKED ISSUE #106: Remove unused jest-spring mock files / folder

### LINKED ISSUE #107: Adjust .yml to make deploy be specific to main

### LINKED ISSUE #113: Cleanup all comments / run linter

### LINKED ISSUE #117: Move types, apicalls, and context providers to helper folder

### PR DESCRIPTION
**Pull request by BlakeDonn**
### What does this PR do?

* Cleans up code considerably 
* Removes all unused imports / files / unnecessary comments
* Adjust `travis.yml` to only deploy to heroku from main branch ( still get CI from every branch, just no false breaking on deploy)
* Moves apiCalls, context providers, and type interfaces to a `helpers` folder for better clarity, also fixes all imports
* Code has been linted for continuity

### Where should the reviewer start?

* Start by checking the format of the code, my linter is kinda weird, if we don't like it we can just roll back to `3b03955` commit

### How should this be manually tested?

* Make sure it still functions locally

### Any background context you want to provide?

* Like I said, the linter is a bit strange, not sure if I like it better or worse, up for input.
* I personally think the `helpers` folder is better for clarity, but if anyone feels otherwise I am down to change it!
* Giving up on memory leak for now, really hard to debug 

### What are the relevant tickets?

closes #76 
closes #106 
closes #107 
closes #117
closes #113

### CODE REVIEW

**Review by timkeresey — APPROVED** [review]
The app link from heroku definitely loads much quicker. All the changes that the linter made work and make sense to me so I think we're good!

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -1,85 +1,101 @@
...
-import {AnswerContext, QuestionContext, ReportContext, AllQuestionFormat} from '../../types'
-import {getQuestions} from '../../apiCalls';
-import {Switch, Route, __RouterContext, Redirect} from 'react-router';
+import React, { useState, useEffect, useContext, useRef } from "react";
+import {
+  AnswerContext,
+  QuestionContext,
+  ReportContext,
+} from "../../helpers/context";
+import { AllQuestionFormat } from "../../helpers/types";
```
Comment (Atos20) [inline src/components/App/App.tsx:7]: I like having a helpers file to import the tyes from.

File: src/components/App/App.tsx [hunk 2]
```diff
@@ -1,85 +1,101 @@
...
-                updateReport={updateReport}/>
-                }/>
-            <Route exact path="/report" component={()=> <Report/>}/>
-            <Route path='/*' component={() => <Error
-              errorMessage={errorMessage} errorNum={errorNum}/>
-              }/>
-          </Switch>
-          </animated.div>
-        ))}
+          <NavBar />
```
Comment (Atos20) [inline src/components/App/App.tsx:64]: I like the new indentation here, I think the look better to me.

File: src/components/Error/Error.test.tsx [hunk 3]
```diff
@@ -1,26 +1,23 @@
...
     render(
       <BrowserRouter>
-        <Error 
-          errorMessage={'here is an error for you'}
-          errorNum={404}
-        />
+        <Error errorMessage={"here is an error for you"} errorNum={404} />
       </BrowserRouter>
-    )
+    );
```
Comment (Atos20) [inline src/components/Error/Error.test.tsx:13]: Was it your linter that changes all the single quotes to double quotes or did you do it manually?

**Review by Atos20 — COMMENTED** [review]
Amazing job creating those extra files, thanks for the hard work.

**Pull request merged**